libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -45,129 +47,139 @@ interface IRushXCommandLineArguments {
 }
 
 export class RushXCommandLine {
-  public static launchRushX(launcherVersion: string, isManaged: boolean): void {
-    RushXCommandLine._launchRushXInternal(launcherVersion, { isManaged });
+  public static launchRushX(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
+    try {
+      const args: IRushXCommandLineArguments = this._getCommandLineArguments();
+      const rushConfiguration: RushConfiguration | undefined = RushConfiguration.tryLoadFromDefaultLocation({
+        showVerbose: false
+      });
+      const eventHooksManager: EventHooksManager | undefined = rushConfiguration
+        ? new EventHooksManager(rushConfiguration)
+        : undefined;
+
+      const ignoreHooks = process.env.INSIDERUSH === '1';
```

Comment:
This generates a little ugliness in the output, because we will see messages about hooks being ignored.  The upside is that we don't have to change the event handler.

Reply:
Rush's environment variable names are always prefixed with `RUSH_` (or `_RUSH_` for internal variables).  They must be documented in [EnvironmentConfiguration.ts](https://github.com/microsoft/rushstack/blob/d18cf333426ba6f31eb38fcef0090c6f2e260a5f/libraries/rush-lib/src/api/EnvironmentConfiguration.ts#L22).  Normally that file handles validating them, however for a performance sensitive operation such as `rushx`, it is okay to read directly from `process.env`.

The name `INSIDERUSH` could be more clear. Looking at the code, it seems that the intent is really to skip event hooks in child processes, because we are already processing a hook?  If so, a better name might be `_RUSH_INVOKING_HOOK` or `_RUSH_SUPPRESS_HOOKS`.  If this is an internal communication mechanism for a specific feature, it's better to make the name highly specific, rather than giving it a general purpose name (that temps other people to rely on it, and then they will get broken if we have to change something).

Reply:
Fixed

Reply:
@nebosite I have renamed this variable to `_RUSH_SUPPRESS_RUSHX_HOOKS` because:

1. It seems to only suppress **rush.json** `eventHooks` for `rushx`, not the other `rush` events
2. The logic (detecting nested invocations) seems potentially tricky, so making it internal will give us more flexibility to adjust the semantics in the future, and
3. End users can use `--ignore-hooks` already, so they probably don't have a strong need for an environment variable anyway.  (If that need arises, we can expose a separate public variable that exactly corresponds to `--ignore-hooks`)

Reply:
OK.

I've been thinking about this some more and I'm not sure we want to use an environment variable this way.   IMO, environment variables are better for communicating context than they are at communicating imperative.  A SUPPRESS_HOOKS variable operates like an imperative, which obscures the reason that we are ignoring the hooks.   It leads to code like this:

```
    if(!process.env["SUPPRESS_HOOKS"]) {
        callhooks();
    }
```
That code looks perfectly correct, but:
1.  it isn't telling me anything about why the hooks are suppressed in the first place.  Any bug is going to be in some distant code that sets the value of SUPPRESS_HOOKS.   
2. This setup also makes it harder for a developer to keep the suppress hooks story correct.  There might be multiple places in the code where a decision is made to suppress hooks or not.  If the decision logic for when to suppress hooks changes, we have to search for all the places we might set it.  
3. It is difficult to write and maintain a cohesive unit test to verify when hooks are called correctly 

For the alternative, consider a context approach with a variable such as IS_RECURSIVE_CALL:
```
    if(!process.env["IS_RECURSIVE_CALL"]) {
        callhooks();
    }
```
Using context gives several improvements:
1.  we can be more certain about the semantics of a conext variable.  It would likely get set right before calling a new process where is is clear from the local context that we are in fact making a recursive call.  
2. Changing the logic for when to set a context variable is less likely to change, and when it does change it is less likely to create a regression because I'm not making a decision about what downstream logic needs to execute.  
3. Easier to read because I see the "why" right next to the hook call.  
4. Easier to unit test.  With one place to add new logic, it is trivial to unit test something like this: 
```
    if(!process.env["IS_RECURSIVE_CALL"]
        && !arguments.ignoreHooks
        && !machineIsInEgypt) {
            callhooks();
    }
```  

I'll leave this unresolved so that you read it.  It's your call to leave the the current code in place, but I think a context variable would be better here.

Reply:
Agreed, the main point of my feedback was that this environment variable is an **_internal implementation detail_** and may be subject to change in the future.  (For example, `rushx` invokes `heft` which runs a Jest test that tries to test the hooks feature itself, but that test fails because the environment variable got inherited into that context. Or `rushx` gets invoked by the `rush build` event hook, which is a different nesting scenario that is NOT suppressed in the current implementation. When exactly is it best to suppress? We need to use the feature a bit before we truly understand all the edge cases.)

Adding the `_` prefix to `_RUSH_SUPPRESS_RUSHX_HOOKS` communicates to Rush users that they should NOT be setting/reading this variable, because its meaning may change, or it may be removed entirely.  That gives us some flexibility to refine it in the future.

You can call the variable whatever makes the most intuitive sense to you, but my suggestion is to make the name very narrow.  (For example `IS_RECURSIVE_CALL` could be narrowed to `_RUSH_RECURSIVE_RUSHX_CALL`.)  Doing so will discourage people from thinking they know what this variable does and disregarding the "internal" designation.

Reply:
OK.  Good name suggestion- I changed it to _RUSH_RECURSIVE_RUSHX_CALL